### ISSUE TITLE
Assets list to show folders first and then files.

### ISSUE THREAD

**Issue by Maksims** [playcanvas/editor#178]
In Code Editor folders are shown first, and then files - this is easy to navigate and looks clean.
In Editor using list view in assets panel, does not shows folders first, it mixes them with files, leading to messy navigation and difference with Code Editor.

Code Editor (good):
![image](https://user-images.githubusercontent.com/636263/95669533-8d5ce580-0b8a-11eb-82f3-831a765d2ed8.png)

Editor (not good):
![image](https://user-images.githubusercontent.com/636263/95669538-96e64d80-0b8a-11eb-9797-93559d696bff.png)

**Comment by vkalpias** [comment]
Hi Max - this is because in the list view rows are sorted alphabetically depending on the sorting column. I thought that when you sort by Name (or any other column) it would be confusing to not see an correctly sorted list because Folders would be special. E.g. if you're trying to see all folders that start with 'g' you can now see it by sorting by Name. 

Also another reason was that you can sort by Type so if you wanted to see all folders you could sort by Type and see all the folders grouped together?

**Comment by willeastcott** [comment]
Well, folders *are* kind of special in Windows Explorer (and will appear before all non-folders in details view). I think the weird thing for me is that we're not being consistent across Code Editor and Editor.

**Comment by vkalpias** [comment]
Just looking at Mac here it seems that it doesn't differentiate folders in terms of order when ordered alphabetically. 
The code editor is using the old system, it's not even using PCUI..

**Comment by Maksims** [comment]
Additionally, after moving to pcui, styling is off, is it something I can make PR to pcui repo and will be incorporated in Editor?

**Comment by vkalpias** [comment]
You can make PRs in pcui - some styling changes are deliberate though so you might wanna point those out :)

**Comment by willeastcott** [comment]
Yeah, just to emphasize that @Maksims, PRs are definitely welcome on the pcui repo. But always open an issue first and give @ellthompson and @vkalpias a chance to respond before submitting the PR. 😄

**Comment by RipcordDev** [comment]
I would personally like to see folders sorted first, then files listed after like in the code editor.  It's the default behaviour in Windows,, which to me makes things more intuitive.  I feel like it used to be this way, but maybe my projects were just smaller in the past and I didn't have as many files in them :D 

I know you can sort by Type but it resets every time I open the project.
